src/com/opc/opceventhandler.h

```diff
@@ -35,11 +38,14 @@ class COpcEventHandler : public CExternalEventHandler, private CThread{
 
     /* functions needed for the external event handler interface */
     void enableHandler() override {
-      start();
+      //nothing call this function
     }
 
     void disableHandler() override {
-      end();
+      if (!isAlive()) {
```

Comment:
in disableHandler at one point you should end the execution of the thread. How would you do that now? Maybe you would like to look at the latest changes in the EventChainExecutionThread.

Reply:
I found disableHandler executed too early and left some commands in the queue when FORTE was killed, so deleting groups (and items) and disconnecting from the server were never get executed. Actually it's strange delayed the end of the thread fix disconnection issue on kill, I left end of the execution in the destructor.

Reply:
I understand that. However I think you need to ensure that when the command queue is cleared that the thread ends. Maybe you do that and I missed it. However if the queue is empty the thread needs to be stopped. 

While I'm writing this I'm thinking if you can't just clear the queue? Do you need to wait for its execution?

Reply:
The execution is important to ensure the client properly unregistered from the server. I have tried to just clear the queue and then remove groups and items from COpcConnection destructor but not always working properly, crash more often.

Reply:
With the last change, basically I did not implement both enableHandler and disableHandler, start and end of the thread controlled by the constructor and the destructor. I use disableHandler only to wake up the thread to empty the command queue.

Reply:
I found this is the only multithreaded process inside FORTE so it may need to be handled differently. Maybe we can only start the process in the constructor and stop it in the destructor and not in between via enableHandler/disableHandler.

Reply:
It is not totally in line with the behavior we expect from an external event handler. But I think for now this is a good solutions.